src/test/rpc/JSONRPC_test.cpp

```diff
@@ -2538,6 +2541,32 @@ class JSONRPC_test : public beast::unit_test::suite
         }
     }
 
+    void
+    testAutoFillNetworkID()
+    {
+        testcase("autofill NetworkID");
+        using namespace test::jtx;
+        Env env{*this, envconfig([&](std::unique_ptr<Config> cfg) {
+                    cfg->NETWORK_ID = 1025;
+                    return cfg;
+                })};
+
+        {
+            Json::Value toSign;
+            toSign[jss::tx_json] = noop(env.master);
+
+            BEAST_EXPECT(!toSign[jss::tx_json].isMember(jss::NetworkID));
+            toSign[jss::secret] = "masterpassphrase";
+            auto rpcResult = env.rpc("json", "sign", to_string(toSign));
+            auto result = rpcResult[jss::result];
+
+            BEAST_EXPECT(!RPC::contains_error(result));
+            BEAST_EXPECT(
+                result[jss::tx_json].isMember(jss::NetworkID) &&
+                result[jss::tx_json][jss::NetworkID] == 1025);
+        }
+    }
```

Comment:
I would also add test where `cfg->NETWORK_ID` is less than 1025, hence `jss::NetworkID` won't be populated (and I guess we get an error from transaction ?)

Reply:
`NETWORK_ID` is defaulted to `0`, so all the existing tests do that already. But I added it explicitly to one of the existing tests.